Added option to install to machine instead of user

MartinGeierino opened the pull request.

mwrock commented on BuildScripts/setup.ps1:
```diff
@@ -1,5 +1,15 @@
+    }
+
+    if( $machineInstall ){
+        $boxstarterPath = Join-Path $env:ProgramFiles Boxstarter
```
Should be ProgramData

mwrock commented on BuildScripts/setup.ps1:
```diff
@@ -1,5 +1,15 @@
+    $machineInstall = $false
+    if( $installArgs -match "MachineInstall" ){
+        $machineInstall = $true
+    }
```
I'd prefer this to be the only option. So we could remove the arg and all checks for `$machineInstall`

mwrock commented:
Thanks a bunch for this PR! I've been wanting to make boxstarter machine local for a while.

MartinGeierino commented:
Hi. 
I made installing to machine the only option and changed the path to ProgramData\Boxstarter and also changed the link paths to CommonDesktopDirectory and CommonStartMenu.
I commented out the part where one byte of the link is changed for another byte because the links weren' t working at first and I couldn't figure out why that byte is changed.

mwrock commented on BuildScripts/setup.ps1:
```diff
@@ -69,7 +70,10 @@ function Create-Shortcut($location, $target, $targetArgs, $boxstarterPath) {
     $lnk.Save()
+    
 
+    #Commted this out because I don't know what it does. Seems to be working without it.
```
Wow I obviously should have commented this because you are right - it is totally unclear what this does. We do want to keep this. It adds a bit to the shortcut link that causes it to open with admin privileges.

mwrock commented:
Sorry for letting this go so long!
